src/Server/AwsCloudFormationTemplates.fs

```diff
@@ -12,481 +9,221 @@ open Shared
     }
 
-    "AWS::ApiGateway::Deployment" => {
-        pulumiType = getPulumiType "AWS::ApiGateway::Deployment"
+    "AWS::ApiGateway::Deployment" => defaultImportIdentity {
         importIdentityParts = ["RestApiId"; "Id"]
         delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
 
-    "AWS::ApiGateway::UsagePlanKey" => {
-        pulumiType = getPulumiType "AWS::ApiGateway::UsagePlanKey"
+    "AWS::ApiGateway::UsagePlanKey" => defaultImportIdentity {
         importIdentityParts = ["UsagePlanId"; "KeyId"]
         delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
+    
+    "AWS::ApplicationAutoScaling::ScalingPolicy" => fun resource data context ->
+        result {
+            let! scalingTargetId = requirePart "ScalingTargetId" data
+            let! policyName = requirePart "PolicyName" data
+            let! importId = 
+                match scalingTargetId.Split("|") with
+                | scalingTargetParts when scalingTargetParts.Length = 3 -> 
+                    scalingTargetParts
+                    |> Array.rev
+                    |> Array.append [| policyName |]
+                    |> String.concat "/" 
+                    |> Ok
+                | _ -> 
+                    Error $"Invalid ScalingTargetId: {scalingTargetId}"
 
-    "AWS::ApplicationAutoScaling::ScalingPolicy" => {
-        pulumiType = getPulumiType "AWS::ApplicationAutoScaling::ScalingPolicy"
-        // as per pulumi registry, import id should be made up of service-namespace , resource-id, scalable-dimension 
-        // and policy-name separated by /
-        // ServiceNamespace, ResourceId and ScalableDimension can be parsed out of ScalingTargetId
-        importIdentityParts = ["ScalingTargetId"; "PolicyName"]
-        delimiter = "/"
-        remapFunc = remapAppScalingPolicy
-        validatorFunc = validateAppScalingPolicy
-    }
+            return importId
+        }
 
-    "AWS::ApplicationAutoScaling::ScalableTarget" => {
-        // as per pulumi registry, import Application AutoScaling Target using the service-namespace , 
-        // resource-id and scalable-dimension separated by /
-        // all three can be parsed from the physical id
-        pulumiType = getPulumiType "AWS::ApplicationAutoScaling::ScalableTarget"
-        importIdentityParts = ["Id"]
-        delimiter = "/"
-        remapFunc = remapAppScalableTarget
-        validatorFunc = validateFromImportIdentityParts
-    }
+    "AWS::ApplicationAutoScaling::ScalableTarget" => fun resource data context ->
+        result {
+            let! physicalId = requirePart "Id" data
+            let! importId = 
+                match physicalId.Split("|") with
+                | parts when parts.Length = 3 -> 
+                    parts
+                    |> Array.rev
```

Comment:
same issue with rev as with ScalingPolicy - format should be `[physicalIdParts[2]; physicalIdParts[0]; physicalIdParts[1]]`

Reply:
Fixed ✅